extensions/azurePublishNew/src/node/azureResources/webApp.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+  ProvisionMethod,
+  ProvisionWorkingSet,
+  ResourceConfig,
+  ResourceDefinition,
+  ResourceProvisionService,
+} from '../provisionService';
+import { AZURE_HOSTING_GROUP_NAME } from '../getResources';
+import { parseRuntimeKey } from '../../../../../Composer/packages/lib/shared';
+
+type WebAppConfig = ResourceConfig & {
+  key: 'webApp';
+  name: string;
+  location: string;
+  resourceGroupName: string;
+};
+
+export type WebAppResult = {
```

Comment:
Nice! I really like how you got type safety around the results and then use the working set Record<>